### DISCUSSION TITLE
Get the ASCII value of pressed  key

### DISCUSSION THREAD

**Discussion by ZeProgFactory** [TolikPylypchuk/SharpHook#49] [Q&A] [answered]
Hi,
Is there a way to get the ASCII value of the key pressed from the `KeyboardHookEventArgs `?
Regards,
Michael

**Comment by TolikPylypchuk** [comment, marked as answer]
Hi! Thanks for asking!

There's a `KeyTyped` event which tries to populate the `KeyChar` property of `KeyboardHookEventArgs` with a Unicode character that was pressed. Note that even though `KeyChar` is a `short`, it should be treated as a `char`. Another thing is that this event is created manually by libuiohook, and not by the OS, so it cannot be suppressed.

If `KeyTyped` doesn't fit your need, then I'm afraid you'll have to create a solution yourself.
